song-server/src/main/java/bio/overture/song/server/service/AnalysisTypeService.java

```diff
@@ -0,0 +1,135 @@
+package bio.overture.song.server.service;
+
+import bio.overture.song.server.model.dto.AnalysisType;
+import bio.overture.song.server.model.entity.AnalysisSchema;
+import bio.overture.song.server.repository.AnalysisSchemaRepository;
+import com.fasterxml.jackson.databind.JsonNode;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.everit.json.schema.Schema;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+import java.util.Set;
+import java.util.function.Supplier;
+
+import static com.google.common.base.Preconditions.checkState;
+import static org.icgc.dcc.common.core.util.stream.Collectors.toImmutableSet;
+import static org.springframework.data.domain.Sort.Direction.ASC;
+import static org.springframework.data.domain.Sort.Direction.DESC;
+import static bio.overture.song.core.exceptions.ServerErrors.ANALYSIS_TYPE_NOT_FOUND;
+import static bio.overture.song.core.exceptions.ServerErrors.MALFORMED_PARAMETER;
+import static bio.overture.song.core.exceptions.ServerException.checkServer;
+import static bio.overture.song.server.model.enums.TableAttributeNames.ID;
+
+@Slf4j
+@Service
+public class AnalysisTypeService {
```

Comment:
You'll need to provide a way to get a single analysis by ID here as related to my comments about RESTful design.